src/components/Authentication/LoginPage.js

```diff
@@ -12,50 +14,80 @@ import Paper from "@material-ui/core/Paper";
 import PasswordInput from "./PasswordInput";
 import TextField from "@material-ui/core/TextField";
 import Typography from "@material-ui/core/Typography";
+import InputAdornment from "@material-ui/core/InputAdornment";
+import EmailOutlinedIcon from "@material-ui/icons/EmailOutlined";
 
-import {makeStyles} from "@material-ui/core/styles";
-import {setToken} from "actions/authActions.js";
+import { makeStyles } from "@material-ui/core/styles";
+import { setToken } from "actions/authActions.js";
 import useAuthStyles from "./styles.js";
+import { ReactComponent as Ellipse1 } from "./loginImages/ellipse1.svg";
+import { ReactComponent as Ellipse2 } from "./loginImages/ellipse2.svg";
+import { ReactComponent as Picture1 } from "./loginImages/picture1.svg";
+import { ReactComponent as Ellipse3 } from "./loginImages/ellipse3.svg";
+import { ReactComponent as Ellipse4 } from "./loginImages/ellipse4.svg";
+import { ReactComponent as Picture2 } from "./loginImages/picture2.svg";
+import { ReactComponent as Picture3 } from "./loginImages/picture3.svg";
+import { ReactComponent as Picture4 } from "./loginImages/picture4.svg";
+import ParentLogin from "./UserLogins/ParentLogin.js";
+import "./LoginPage.scss";
 
 const useStyles = makeStyles((theme) => ({
-    "forgot": {
-        "color": theme.colors.darkGray,
+    forgot: {
+        color: theme.colors.darkGray,
         "text-decoration": "none",
     },
-    "smallerRoot": {
+    smallerRoot: {
         [theme.breakpoints.up("sm")]: {
-            "width": "412px",
+            width: "412px",
         },
     },
 }));
 
 const LOGIN = gql`
-    mutation Login($password: String!, $username: String!) {
-        tokenAuth(password: $password, username: $username) {
-            token
-            payload
-        }
+  mutation Login($password: String!, $username: String!) {
+    tokenAuth(password: $password, username: $username) {
+      token
+      payload
     }
+  }
+`;
+
+const GET_USER_TYPE = gql`
+  query GetUserType($username: String!) {
+    userType(userName: $username)
+  }
 `;
 
 const LoginPage = () => {
     const history = useHistory();
-    const {state} = useLocation();
+    const { state } = useLocation();
     const dispatch = useDispatch();
-    const token = useSelector(({auth}) => auth.token);
-
+    const token = useSelector(({ auth }) => auth.token);
+    const [userType, setUserType] = useState("");
     const [email, setEmail] = useState(state?.email);
+    const [realEmail, setRealEmail] = useState("");
```

Comment:
I need 2 values since there is a bug when you click the login button before you type in an email, itll move it to the next page as soon as you finish the email, without clicking the button. this acts as a buffer between the 2 because realemail is not affected immediately after the state is set by the emailfield.

Reply:
Can't you add a check to make sure the email isn't empty when the button is clicked?